Add initial README describing the project (#3)
bisdn/docker-yocto-builder: README.md

# Yocto Build Environment as Docker Container

> This repository aims to provide a complete yocto build environment for building

Reviewer: ```suggestion
This repository aims to provide a complete Yocto build environment for building
```

BISDN Linux in a docker container.

## Build Environment

> The build environment is based on the recommendations documented in the yocto
> project for Ubuntu and Debian

Reviewer: ```suggestion
The build environment is based on the recommendations documented in the Yocto
Project for Ubuntu and Debian
```

[here](https://docs.yoctoproject.org/3.1.7/ref-manual/ref-system-requirements.html#ubuntu-and-debian).
In addition to the packages mentioned there, it also includes
[libelf](https://directory.fsf.org/wiki/Libelf) with development headers and

> [PyYAML](https://pyyaml.org), since those are needed to build some of the

Reviewer: Very good explanation and documentation

packages included in BISDN Linux.
Last but not least, the environment also includes
['repo'](https://android.googlesource.com/tools/repo), a tool developed by google
to manage multiple git repositories and their respective versions from one
manifest file. Since 'repo' requires python and building Open Network Linux
depends on python2, the environment includes both python3, as well as python2.

In addition to providing all packages and dependencies needed for building BISDN

> Linux with the yocto build chain, the environment already includes a

Reviewer: ```suggestion
Linux with the Yocto build chain, the environment already includes a
```

minimalistic [.gitconfig](./gitconfig) needed when using the aforementioned

> 'repo' tool and sets the locale to 'en_US.UTF-8'. Since the yocto build tool

Reviewer: ```suggestion
'repo' tool and sets the locale to 'en_US.UTF-8'. Since the Yocto build tool
```

'bitbake' does not allow building as 'root' user, the environment includes a

> 'builder' user, which is used by default when running the '-ci' version of the

Reviewer: ```suggestion
'builder' user, which is used by default when running the 'ci' version of the
```
Since we mention 'base' and 'ci' targets below, should we drop the dash (-) here?

Author: done

> container.  All details on how the container is build and which packages are

Reviewer: ```suggestion
container.  All details on how the container is built and which packages are
```

Author: done

included can be found directly in the [Dockerfile](./Dockerfile) included in
this repository.

The Dockerfile used to build the environment described above is written as a
[multi-stage build file](https://docs.docker.com/build/building/multi-stage/)
and includes the two targets 'base' and 'ci'. The only difference between these
two targets is the `USER` which the container will use to run all commands by
default. The 'ci' target is intended to be ready to use for any pipelines, since
it automatically runs all commands as `builder` user and therefore allows to
directly get started with 'bitbake'. The 'base' target on the other hand will by
default execute everything as 'root' and is intended to allow easier

> modification during runtime (like installing more packages or changing and

Reviewer: Giving examples of what one might modify is great, and makes the README better, simpler and nicer to read.

adding users).

## GitHub workflows

This project uses
[GitHub workflows](https://docs.github.com/en/actions/using-workflows/about-workflows)

> to automatically build the aforementioned environment as container. All

Reviewer: ```suggestion
to automatically build the aforementioned environment as a container. All
```
or maybe 'in a' container? Not sure if they are better suggestions though. Feel free to pick and choose.

Author: done

containers are then published as
[GitHub Packages](https://github.com/orgs/bisdn/packages?repo_name=docker-yocto-builder).
To ensure consistent code quality in all files in this repository, the
[GitHub Super-Linter](https://github.com/github/super-linter) is run on every
pull request for all files changed.
In addition to that, all commits are required to follow the
[conventional commit format](https://www.conventionalcommits.org) and an
automated commit linter for this format is run on every pull request.
For detailed information on all workflows, please refer to the workflow files
included in the [.github/workflows/](./.github/workflows/) folder in this
repository.

## Contributing

If you feel like contributing to this project, please use any of the GitHub
tools available and choose whatever fits your contribution best. We welcome pull
requests as well as bug reports and feature request in form of issues. If you
want to contribute in any other way or style, please open an issue and let us

> know.

Reviewer: +1
This is something we don't do enough in my opinion, and gives a positive message to those thinking about contributing.
